pkg/aggregator/aggregator.go

```diff
@@ -70,6 +70,20 @@ func (s *Stats) add(stat int64) {
 	s.LastFlush = stat
 }
 
+func (s *Stats) copy() *Stats {
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	stats := &Stats{
+		Flushes:    s.Flushes,
+		FlushIndex: s.FlushIndex,
+		LastFlush:  s.LastFlush,
+		Name:       s.Name,
+	}
+	copy(stats.Flushes[:], s.Flushes[:])
+	return stats
```

Comment:
TIL ! I thought arrays behaved like slices, and the `Flushes: s.Flushes` was not on purpose 🤦
I can't just do `stats := *s` because it complains that it copies the `Mutex`, but at least I can get rid of the `copy` 👍